core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -759,7 +759,13 @@ public void releaseSnapshot(IndexCommit snapshot) throws IOException {
      */
     public void failShard(String reason, @Nullable Throwable e) {
         // fail the engine. This will cause this shard to also be removed from the node's index service.
-        getEngine().failEngine(reason, e);
+        final Engine engine = getEngineOrNull();
```

Comment:
haha. I fixed this inline - forgot :) I think this should be a separate PR - it's not really needed for this one (we protect for it).